### PR TITLE
Fix E2E test setup.

### PR DESCRIPTION
**Pull request by prafullbhosale**
- Moved sample test app to RepoRoot/TestApps/*
- Skip TestApps directory during restore.
- Restore TestApps during actual test.

cc @pranavkm

### CODE REVIEW

File: TestApps/Nuget.config [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    <add key="local" value="..\\artifacts\\build" />
```
Comment (pranavkm) [inline TestApps/Nuget.config]: Use forward slashes. This wouldn't work on xplat.

**Comment by prafullbhosale** [comment]
Additional tests need changes to run without src folder in place.

**Comment by prafullbhosale** [comment]
:up: :date: 
Tests pass after removing src folder as well.

File: test/Microsoft.VisualStudio.Web.CodeGeneration.Templating.Test/RazorTemplatingTests.cs [hunk 2]
```diff
@@ -53,15 +53,22 @@ public class RazorTemplatingTests
 
         private ICompilationService GetCompilationService()
         {
-            ProjectContext context = CreateProjectContext(null);
-            var applicationInfo = new ApplicationInfo("", context.ProjectDirectory);
+            var originalAppBase = Directory.GetCurrentDirectory();
+            var testAppName = "ModelTypesLocatorTestClassLibrary";
+#if NET451
+            var testAppPath = Path.GetFullPath(Path.Combine(originalAppBase, "..","..","..","..","..", "..","TestApps", testAppName));
```
Comment (pranavkm) [inline test/Microsoft.VisualStudio.Web.CodeGeneration.Templating.Test/RazorTemplatingTests.cs]: You could copy the code from MusicStore to locate the sln root

Comment (pranavkm) [inline test/Microsoft.VisualStudio.Web.CodeGeneration.Templating.Test/RazorTemplatingTests.cs]: https://github.com/aspnet/Mvc/blob/dev/test/Microsoft.AspNetCore.Mvc.TestCommon/SolutionPathUtility.cs

**Pull request merged**